include/directx/d3dx12.h

```diff
@@ -4217,13 +4239,7 @@ class CD3DX12FeatureSupport
     // 7: Shader Model
     D3D_SHADER_MODEL HighestShaderModel(D3D_SHADER_MODEL RequestedShaderModel = D3D_HIGHEST_SHADER_MODEL)
     {
-        if (RequestedShaderModel <= m_dShaderModel.HighestShaderModel) {
-            return RequestedShaderModel;
-        }
-
-        D3D12_FEATURE_DATA_SHADER_MODEL dShaderModel;
-        m_pDevice->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &dShaderModel, sizeof(D3D12_FEATURE_DATA_SHADER_MODEL));
-        return dShaderModel.HighestShaderModel;
+        return min(RequestedShaderModel, m_dShaderModel.HighestShaderModel);

```

Comment:
The reason the D3D API for these uses an in-out pattern for this is because:
1. The app compiled with one set of enum values for feature level
2. D3D can be updated to include new enum values
3. If D3D returns new enum values to an app that doesn't understand those values, the app is seeing data that it doesn't necessarily know how to handle

For `d3dx12.h`, since this is compiled into the app, I think we can just use getters, instead of passing in a requested shader model. The value returned from the getter is *guaranteed* to be one of the values that the app knew about when it was compiled.